code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -470,30 +485,73 @@ func getBlobbersForRequest(request newAllocationRequest, edb *event.EventDb, bal
 //  400:
 func (srh *StorageRestHandler) getCollectedReward(w http.ResponseWriter, r *http.Request) {
 	var (
-		startBlock, _ = strconv.Atoi(r.URL.Query().Get("start_block"))
-		endBlock, _   = strconv.Atoi(r.URL.Query().Get("end_block"))
-		clientID      = r.URL.Query().Get("client_id")
+		startBlock, _    = strconv.Atoi(r.URL.Query().Get("start-block"))
+		endBlock, _      = strconv.Atoi(r.URL.Query().Get("end-block"))
+		clientID         = r.URL.Query().Get("client-id")
+		startDateString  = r.URL.Query().Get("start-date")
+		endDateString	 = r.URL.Query().Get("end-date")
+		dataPointsString = r.URL.Query().Get("data-points") 
 	)
 
+	var dataPoints int64
+	dataPoints, err := strconv.ParseInt(dataPointsString, 10, 64)
+	if err != nil {
+		dataPoints = 1
+	}
+
+
 	query := event.RewardQuery{
 		StartBlock: startBlock,
 		EndBlock:   endBlock,
 		ClientID:   clientID,
+		DataPoints: dataPoints,
 	}
 	edb := srh.GetQueryStateContext().GetEventDB()
 	if edb == nil {
 		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
 		return
 	}
-	collectedReward, err := edb.GetRewardClaimedTotal(query)
-	if err != nil {
-		common.Respond(w, r, 0, common.NewErrInternal("can't get rewards claimed", err.Error()))
-		return
+
+
+	if startBlock > 0 && endBlock > 0 {
```

Comment:
Check the invalid block range and return error, such as:
- startBlock > endBlock
- block number < 0

Also, we need to do pagination.

Reply:
https://github.com/0chain/0chain/pull/1475 has some local functions to do this:
https://github.com/0chain/0chain/blob/1210e4b632f1d08416a3402f755b5b85b685b0a1/code/go/0chain.net/smartcontract/storagesc/handler.go#L272

Reply:
we probably should limit the number (100 i think) of points and be fine without pagination

Reply:
@lawRathod does the data-points work for blocks, I find the comment says the `data-points` only support dates.

Reply:
I did add them for dates only but it seemed easy enough to include it for blocks too, so done that!